src/content/content_manager.cc

```diff
@@ -619,13 +624,12 @@ void ContentManager::_rescanDirectory(std::shared_ptr<AutoscanDirectory>& adir,
     if (location.empty()) {
         log_error("Container with ID {} has no location information", containerID);
         return;
-        //        continue;
-        // throw_std_runtime_error("Container has no location information");
     }
 
-    DIR* dir = opendir(location.c_str());
-    if (!dir) {
-        log_warning("Could not open {}: {}", location.c_str(), std::strerror(errno));
+    std::error_code ec;
+    auto dirEnt = fs::directory_entry(location);
```

Comment:
The constructor thows on error, I guess passing the ec and checking it would help.

Reply:
Checking of entry is done already, so I don't check the constructor return separately